internal/pkg/renderer/renderer.go

```diff
@@ -119,8 +124,8 @@ func (r *Renderer) Render(p *pack.Pack, variables map[string]interface{}) (*Rend
 // RenderOutput performs the output template rendering.
 func (r *Renderer) RenderOutput() (string, error) {
 
-	// If we don't have a template file, then return early.
-	if r.pack.OutputTemplateFile == nil {
+	// If we don't have a template file or any aux files, return early.
+	if r.pack.OutputTemplateFile == nil && r.pack.AuxiliaryFiles == nil {
```

Comment:
```suggestion
	if r.pack.OutputTemplateFile == nil && len(r.pack.AuxiliaryFiles) == 0 {
```

I think an empty slice is not `nil`. But also, it doesn't seem like `r.pack.AuxiliaryFiles` is used here, so maybe we don't need this check?

Reply:
It does make sense, but adding this check does the opposite of that when `len(r.pack.AuxiliaryFiles) > 0` and `r.pack.OutputTemplateFile == nil`. Nothing inside this function cares about `r.pack.AuxiliaryFiles` so running it just because there are auxiliary files is unnecessary when `r.pack.OutputTemplateFile` is `nil`. 

But worse, looking at it again, this change actually introduces a `nil` panic in this scenario:

```
$ nomad-pack render  --render-output-template .

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x1031e9dac]

goroutine 1 [running]:
github.com/hashicorp/nomad-pack/internal/pkg/renderer.(*Renderer).RenderOutput(0x140006915c0)
        /Users/laoqui/go/src/github.com/hashicorp/nomad-pack/internal/pkg/renderer/renderer.go:132 +0x3c
github.com/hashicorp/nomad-pack/internal/pkg/manager.(*PackManager).ProcessOutputTemplate(...)
        /Users/laoqui/go/src/github.com/hashicorp/nomad-pack/internal/pkg/manager/manager.go:118
github.com/hashicorp/nomad-pack/internal/cli.(*RenderCommand).Run(0x1400058e540, {0x140000300a0?, 0x0?, 0xffffffffffffffff?})
        /Users/laoqui/go/src/github.com/hashicorp/nomad-pack/internal/cli/render.go:232 +0x728
github.com/mitchellh/cli.(*CLI).Run(0x140000003c0)
        /Users/laoqui/go/pkg/mod/github.com/mitchellh/cli@v1.1.4/cli.go:262 +0x4a8
github.com/hashicorp/nomad-pack/internal/cli.Main({0x14000030080?, 0x4, 0x4})
        /Users/laoqui/go/src/github.com/hashicorp/nomad-pack/internal/cli/main.go:93 +0x268
main.main()
        /Users/laoqui/go/src/github.com/hashicorp/nomad-pack/main.go:12 +0x90
```

Reply:
🤦 https://github.com/hashicorp/nomad-pack/pull/310🤦